lib/active_merchant/billing/gateways/trust_commerce.rb

```diff
@@ -250,6 +250,7 @@ def void(authorization, options = {})
 
       def verify(credit_card, options = {})
         add_creditcard(options, credit_card)
+        options[:order_id] = nil
```

Comment:
This change doesn't seem like it'll do anything because nothing ever happens with the `options` hash after `add_creditcard` or in the `commit` method, unless I'm missing something.

Reply:
According to the tests, the gateways rejects a field that is added by default in the library (order_id) so it needs to be removed for the verify in order to complete the verify requests successfully. If you need further clarification please let me know.

Reply:
Oh I see here. So, it looks like you are using the `options` instead of the `parameters` (what would normally be `post`).

You should create a `parameters` hash and pass that to the `add_creditcard` method, instead of the `options` hash. That hash is going to be passed to the method from the user and AM takes the values from that hash and constructs the payload that we send to the gateway API.

It also doesn't look like `add_creditcard` adds any `order_id` to the payload, so you probably won't need to delete that value.